pkg/admission/admission_controller.go

```diff
@@ -197,6 +199,7 @@ func (c *AdmissionController) processPod(req *admissionv1.AdmissionRequest, name
 
 	if c.shouldLabelNamespace(namespace) {
 		patch = updateLabels(namespace, &pod, patch)
+		patch = c.updatePreemptionInfo(&pod, patch)
```

Comment:
I think some refactor or clean-up is becoming unavoidable in this file. It's getting larger and these functions are inconsistent (there are more). Not in scope, just saying, maybe it's worth a follow-up JIRA.

Reply:
Agreed.